### PR TITLE
Feature/create fundraisers page

### PR DESCRIPTION
**Pull request by Mhmdabed11**
# Describe your PR
Create the fundraisers page
Related to #289 

## Pages/Interfaces that will change
A new fundraisers page will be created

## Steps to test

1. Navigate to /fundraisers

### CODE REVIEW

**Comment by Mhmdabed11** [comment]
@magnificode  I have some questions regarding this PR so it can be 100% done from my side:
1. Is there a way to filter the businesses which have a donationLink entry using the REST API. I cannot fins such one in the api docs.
2. image
For the above image, what should happen when we press on the button? And also I need the public ID of the background image if possible.
3.For this page, I think it makes sense to have more than 12 entries in the grid since the size of each gridItem is relatively small. This means that we need to edit the useSearch hook to accept an optional pageSize argument. I can do this if we agree to do it.

File: src/components/Cards/FundraiserCard.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import React, { forwardRef, useRef } from 'react';
+import { Box, Icon, Text, useDisclosure, useTheme } from '@chakra-ui/core';
+import PropTypes from 'prop-types';
+import {
+  CardButton,
+  CardButtonGroup,
+  CardContent,
+  CardHeading,
+  CardText,
```
Comment (magnificode) [inline src/components/Cards/FundraiserCard.js:9]: We can remove this import if CardText is not used :)

File: src/components/Cards/FundraiserCard.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            fontSize={theme.fontSizes.xl}
+            lineHeight="1"
+            overflowWrap="break-word"
+            wordWrap="break-word"
+            wordBreak="break-word"
+            hyphens="auto"
+          >
+            {name}
+          </CardHeading>
+          {/* {location && (
```
Comment (magnificode) [inline src/components/Cards/FundraiserCard.js:52]: Remove commented code here

**Comment by magnificode** [comment]
> @magnificode I have some questions regarding this PR so it can be 100% done from my side:
> 
> 1. Is there a way to filter the businesses which have a donationLink entry using the REST API. I cannot fins such one in the api docs.
> 2. image
>    For the above image, what should happen when we press on the button? And also I need the public ID of the background image if possible.
>    3.For this page, I think it makes sense to have more than 12 entries in the grid since the size of each gridItem is relatively small. This means that we need to edit the useSearch hook to accept an optional pageSize argument. I can do this if we agree to do it.

Excellent questions @Mhmdabed11 

1. We should be able to, maybe @elchris can shed some light there?
2. Public ID for that image: `coffee-shop_rbgwyx` - @shapirone, could you speak to the functionality for that CTA?
3. It looks like the designs have 14 items output on this page. I think adding that optional argument would be excellent!

Last thing, would you mind removing the "Home" link from the nav? The additional fundraisers page makes the nav a bit long.

Screen Shot 2020-08-20 at 11 31 32 AM

Based on the design, the order should be as follows:

Screen Shot 2020-08-20 at 11 40 15 AM


Thank you so much for the hard work on this @Mhmdabed11 - this is looking great!

**Comment by Mhmdabed11** [comment]
@magnificode  Thank you for answering my questions and you are most welcome !

**Comment by elchris** [comment]
oh my bad to get businesses raising funds, pass:

in-need=true

It's been documented on the API Doc:

https://developer.rbb-api.com/doc

click on the 3rd endpoint:

GET /api/v1/businesses

@Mhmdabed11 @magnificode

**Comment by Mhmdabed11** [comment]
@elchris thank you for your reply. I actually did read this in the docs, but I ended up with some businesses with `donationLink:null` while setting `in-need=true`.

**Comment by shapirone** [comment]
@Mhmdabed11 sorry I wasn't clear but since we support fundraisers that aren't directly attached to businesses, we have a different (and much simpler) UX for adding a fundraisers vs businesses. Designs can be found in [our Figma.](https://www.figma.com/file/Lo1ZcBOO9t4QRbX2I0PwBQ/RBB?node-id=1983%3A8264)

@magnificode do we have a separate issue for creating the Fundraiser form or should that be handled as part of this ticket?

**Comment by magnificode** [comment]
@shapirone @Mhmdabed11 lets focus on displaying the existing businesses with donation links in this PR.

We'll create another task for submitting a fundraiser not connected to a business.

**Comment by magnificode** [comment]
@Mhmdabed11 this looks awesome, and great touch pre-checking the "Has Fundraiser" checkbox when submitting from this page. I think that this will work great for now, we can iterate on the data structure on the backend further down the line if we want to split this out into it's own table.

The final, tiny thing, if you wouldn't mind @Mhmdabed11 - could we add a couple of transforms to this image? As it stands, the image itself makes the text on top a bit tough to read.

Screen Shot 2020-08-24 at 10 39 23 AM

In order to match the design, I added a couple of transformations to the image on cloudinary:

```
background: "#000000"
effect: "saturation:-78"
opacity: "47"
```

The Cloudinary Image takes a `transformations` object as props, which you should be able to pass these values to, in order to see the result, which should look similar to this: https://res.cloudinary.com/rebuild-black-business/image/upload/b_rgb:000000,e_saturation:-78,o_47/v1597945031/assets/coffee-shop_rbgwyx.jpg

Let me know if you're able to get that to work!

Thank you again for all the hard work! Once that image is handled we can get this merged in :)

**Comment by magnificode** [comment]
Awesome work @Mhmdabed11 merging this in now!

**Pull request merged**